src/ir/context.rs

```diff
@@ -697,6 +702,19 @@ impl<'ctx> BindgenContext<'ctx> {
             CXType_Float => TypeKind::Float(FloatKind::Float),
             CXType_Double => TypeKind::Float(FloatKind::Double),
             CXType_LongDouble => TypeKind::Float(FloatKind::LongDouble),
+            CXType_Complex => {
+                let float_type = ty.elem_type()
+                    .expect("Not able to resolve complex type?");
+                let float_kind = match float_type.kind() {
+                    CXType_Float => FloatKind::Float,
+                    CXType_Double => FloatKind::Double,
+                    CXType_LongDouble => FloatKind::LongDouble,
+                    _ => panic!("Non floating-type complex?"),
+                };
+
+                self.saw_complex.set(true);
```

Comment:
Hmm... Probably we should only output it if we try to convert a rust type to a complex. It's likely, over all if you use whitelisting, that we end up parsing it but not generating output for it. Could you add a method to set this flag, and call it from `to_rust_ty`, where we generate the `__BindgenComplex` instead?

Note that to modify a `Cell` the context doesn't need to be mutable, so we can set it during codegen.